src/hotspot/share/runtime/threads.cpp

```diff
@@ -548,7 +548,10 @@ jint Threads::create_vm(JavaVMInitArgs* args, bool* canTryAgain) {
   // Initialize global modules
   jint status = init_globals();
   if (status != JNI_OK) {
-    main_thread->smr_delete();
+    if (!main_thread->has_pending_exception()) {
+      main_thread->smr_delete();
+    }
```

Comment:
We have the exact same problem below after the call to init_globals2 (very recently introduced by JDK-8240774).

Reply:
Thanks @kimbarrett I had obviously missed that.

Reply:
Fixed.

Reply:
Is it possible to have a pending exception after init_globals() now since Java initializations are done in init_globals2?